aibolit/patterns/supermethod/supermethod.py

```diff
@@ -20,27 +35,16 @@ def value(self, filename: str):
         """
         results = []
         source_code = read_text_with_autodetected_encoding(filename)
-        tree = parse(source_code)
         lines = source_code.splitlines()
-        for _, method_decl_node in tree.filter(MethodDeclaration):
-            code_line = method_decl_node.position.line
-            for _, super_method_inv in method_decl_node.filter(SuperMethodInvocation):
+        tree = JavaPackage(filename)
+        for method_decl_node in tree.nodes_by_type(ASTNodeType.METHOD_DECLARATION):
+            code_line = tree.get_attr(method_decl_node, 'source_code_line')
+            for super_method_inv in tree.nodes_by_type(ASTNodeType.SUPER_METHOD_INVOCATION):
```

Comment:
Is it something with me, but always when I see `inv` I think about some sort of inverse. And by the way, you cannot declare super method, it is already declared in class parent, so, basically, the only thing you can do with super method is to invoke them.

Reply:
Hah, I decided not to chan original names:)